### PR TITLE
feat(new-agent): license scanner for SPDX-License-Identifier

### PR DESCRIPTION
**Pull request by ag4ums**
This new license scanner identifies `SPDX-License-Identifier:   <license-name>` license information from  from files.

### CODE REVIEW

File: src/ojo/Makefile [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Copyright Siemens AG 2017-2018
```
Comment (mcjaeger) [inline src/ojo/Makefile]: no !!! :-D

**Comment by rlintu** [comment]
Do we need a new agent for this? nomos can be enhanced to detect those license identifiers, actually I've done that for few licenses (and some other corrections https://github.com/fossology/fossology/issues/786)

**Comment by rlintu** [comment]
Sorry, wrong button pressed ...

**Comment by mcjaeger** [comment]
yes we do, because it should enable auto conclusion when the SPDX license identifier is matched. Opposed to Nomos regex matching we could do:

* Ojo finds a SPDX-id-based license statement
* Nomos does not find anything else

-> auto conclude and done.

**Comment by mcjaeger** [comment]
some projects to test with:

https://github.com/ARM-software/arm-trusted-firmware
https://github.com/ARM-software/CMSIS_5
https://github.com/zephyrproject-rtos/zephyr

more here:
https://wiki.spdx.org/view/Business_Team/Adoption#Use_of_SPDX_Short_Identifiers

**Comment by mcjaeger** [comment]
for the decider, to conclude:
* ojo alone should be OK, no other agent finds something
* if other agents find something different, then no decision

**Comment by mcjaeger** [comment]
Just came across this: how about adding a few words to the ojo license analysis in the upload form just as it is done for monk and nomos? I think users will not understand what ojo does otherwise.

**Comment by mcjaeger** [comment]
I am missing the decider part?

**Comment by mcjaeger** [comment]
I think Ojo is missing in the license list (just see nomos monk and ninka there)

File: src/ojo/agent/ojos.cc [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+ * @brief Entry point for ojo agent
+ * @page ojo OJO Agent
+ * @tableofcontents
+ *
+ * OJO agent uses regular expressions to find out SPDX License identifiers
+ * from a file.
+ *
+ * The agent runs in multi-threaded mode and creates a new thread for
+ * every pfile for faster processing.
+ *
```
Comment (mcjaeger) [inline src/ojo/agent/ojos.cc]: nice! command line capability!

**Comment by mcjaeger** [comment]
this is going to be cool:

Screen Shot 2019-08-01 at 20 34 17

and this:

Screen Shot 2019-08-01 at 20 34 44

File: src/ojo/agent/OjosDatabaseHandler.cc [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+unsigned long OjosDatabaseHandler::saveLicenseToDatabase(
+    OjoDatabaseEntry &entry) const
+{
+  QueryResult queryResult = dbManager.execPrepared(
+    fo_dbManager_PrepareStamement(dbManager.getStruct_dbManager(),
+      "ojoInsertLicense",
+      "INSERT INTO license_file"
+      "(rf_fk, agent_fk, pfile_fk, rf_match_pct)"
+      " VALUES($1,$2,$3,100) RETURNING fl_pk",
+      long, long, long),
```
Comment (ag4ums) [inline src/ojo/agent/OjosDatabaseHandler.cc]: 100 is not require as Ojo don't find any percentage

Comment (GMishx) [reply src/ojo/agent/OjosDatabaseHandler.cc]: Removed, please check.

File: debian/control [hunk 4]
```diff
@@ -236,6 +236,18 @@ Description: architecture for analyzing software, ninka
  .
  This package contains the ninka wrapper agent programs and their resources.
 
+Package: fossology-ojo
+Architecture: any
+Depends: fossology-common, ${shlibs:Depends}, ${misc:Depends}
+Description: architecture for analyzing software, ninka
+ The FOSSology project is a web based framework that allows you to
```
[the remark this answers is not in the document]
Comment (GMishx) [reply debian/control]: Fixed, thanks for pointing out.

**Pull request merged**